server/handlers/order.js

```diff
@@ -77,3 +78,22 @@ exports.getOrdersByClientID = async function (req, res, next) {
   const orders = result.map((element) => Order.fromMongoJSON(element));
   return res.json(orders);
 };
+
+exports.completeOrderValidator = [orderIDParamValidator];
+
+exports.completeOrder = async function (req, res, next) {
+  // Insert the new order
+  var result;
+  try {
+    result = await dao.completeOrder(req.params.orderID);
+  } catch (err) {
+    console.error(`CompleteOrder() -> couldn't patch the order: ${err}`);
+    return res.status(500).end();
+  }
+  if (!result.value) {
+    console.error(`CompleteOrder() -> couldn't find the requested order`);
+    return res.status(400).end();
```

Comment:
This should be a 404

Reply:
In the API there is no error 404. Also, I asked the API team and I was told to use 400 instead of 404 since it was a bad request from the user to update a status for a non-existing order.